web/app/js/components/TopRoutes.jsx

```diff
@@ -113,49 +112,102 @@ class TopRoutes extends React.Component {
 
   renderRoutesQueryForm = () => {
     return (
-      <Grid container direction="column">
-        <Grid item container spacing={8} alignItems="center">
-          <Grid item xs={6} md={3}>
-            { this.renderTextInput("Service", "resource_name", "Name of the configured service") }
-          </Grid>
-          <Grid item xs={6} md={3}>
-            { this.renderTextInput("Namespace", "namespace", "Namespace of the configured service") }
-          </Grid>
-        </Grid>
+      <CardContent>
+        <Grid container direction="column">
+          <Grid item container spacing={8} alignItems="center">
+            <Grid item xs={6} md={3}>
+              { this.renderNamespaceDropdown("Namespace", "namespace", "Namespace of the configured service") }
+            </Grid>
 
-        <Grid item container spacing={8} alignItems="center">
-          <Grid item xs={6} md={3}>
-            { this.renderTextInput("From", "from_name", "Resource name") }
-          </Grid>
-          <Grid item xs={6} md={3}>
-            { this.renderTextInput("From type", "from_type", "Resource type") }
+            <Grid item xs={6} md={3}>
+              { this.renderServiceDropdown() }
+            </Grid>
           </Grid>
-          <Grid item xs={6} md={3}>
-            { this.renderTextInput("From namespace", "from_namespace", "Resource namespace") }
+
+          <Grid item container spacing={8} alignItems="center">
+            <Grid item xs={6} md={3}>
+              { this.renderTextInput("From", "from_name", "Resource name") }
```

Comment:
could/should these 3 text inputs be reworked into 2 fields, `From Namespace` and `From Resource`, similar tap/top forms? also, add autocomplete?

![screen shot 2018-11-29 at 10 03 00 am](https://user-images.githubusercontent.com/236915/49241924-136c4a80-f3be-11e8-8987-631fa75e91ac.png)

Reply:
yes I think so! will delegate that to a polish followup

Reply:
filed #1894